core/arch/riscv64/riscv64.asm

```diff
@@ -264,23 +264,34 @@ GLOBAL_LABEL(_dynamorio_runtime_resolve:)
  */
         DECLARE_FUNC(dynamorio_clone)
 GLOBAL_LABEL(dynamorio_clone:)
-/* FIXME i#3544: Not implemented */
+        addi     ARG2, ARG2, -16 /* Description: newsp = newsp + 16. */
+        sd       ARG1, 8 (ARG2)
+        sd       ARG6, 0 (ARG2) /* The func is now on TOS of newsp. */
+        li       SYSNUM_REG, SYS_clone /* All args are already in syscall registers.*/
+        ecall
+        bnez     ARG1, dynamorio_clone_parent
+        ld       ARG1, 0 (sp)
+        ld       ARG2, 8 (sp)
```

Comment:
So we load the flags back here into ARG2. Is that needed? Is it part of the `dynamorio_clone` convention?

Reply:
On AArch64, `ldp` and `stp` cannot push a single register, so I think ARG1(x0) was pushed just to build a pair of registers. Also in the x86-64 version I don't see ARG1(rdi) being used at all (except being forwarded to SYS_clone).

Reply:
Fixed https://github.com/DynamoRIO/dynamorio/pull/6063/commits/63ded0ec125b1ac93c7264beff447937d669ec19